### PR TITLE
feat: my schedule filter

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/5763

Screenshot from 2021-02-14 08-54-30
Screenshot from 2021-02-14 08-54-41
Screenshot from 2021-02-14 09-00-02

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mariobehling** [comment]
Which issue does it solve?

Where is the button to see "My schedule"?

**Comment by maze-runnar** [comment]
> Where is the button to see "My schedule"?

In the rightside, in the screenshot, above tracks.

**Comment by maze-runnar** [comment]
> Which issue does it solve?

Adding my Schedule filter to schedule.

**Comment by mariobehling** [comment]
Please work according to best practices. The issue that I see comes closest to this one https://github.com/fossasia/open-event-frontend/issues/5763

Here we discussed making a dedicated page. If we change the requirement to filters on the same page you need to mention this and a decision has to be made. Follow-up PRs are open which enable attendees to share their schedule publicly. I am not sure if the current approach would enable this.

@iamareebjamal Please help to clarify.

**Comment by mariobehling** [comment]
Please match a PR with an issue always. We are loosing track what is going on and what development decision has been made.

**Comment by iamareebjamal** [comment]
@mariobehling A separate page will duplicate the code a lot and will be significantly difficult to implement than adding a filter. By all intents and purposes, the filtered page is a separate page in itself as it has a separate link that can be copied and pasted, and it also provides features to change the timezone and search and filter according to date and tracks

If let's say we only show favourite sessions in a separate page, without timezone switcher and any other filter feature, even without the date grouping, even then it would take significant effort and duplication to implement. What benefit does implementing a separate page give us?

**Comment by iamareebjamal** [comment]
> Follow-up issues are open which enable attendees to share their schedule publicly. I am not sure if the current approach would enable this.

A different page would also not enable this as there is no feature currently which enables other users to see favourites of other users, this requirement was added after the API was finalized as far as I know. This feature needs more implementation discussion as I don't know how it would be possible in current API structure. Either we can make all user's favourite public or all user's favourite private (this is default for now), as there is no option to store different setting for user currently

**Comment by mariobehling** [comment]
Thanks @iamareebjamal. Let's discuss this in the weekly meeting, but I see an increasing tendency that our issues and what is done in PRs divert. There are a number of issues regarding the my schedule feature. I have labeled the most important ones high priority: https://github.com/fossasia/open-event-frontend/issues?q=is%3Aopen+label%3A%22Priority%3A+High%22+my+schedule

Attendees of virtual events say they want to be able to see what others are doing, hence in November I opened an issue for the feature to enable attendees to share their schedule and attendance in other sessions. We are simply following existing solutions like sched.

**Comment by iamareebjamal** [comment]
We'll be able to show which people will attend a session like here - https://github.com/fossasia/open-event-frontend/issues/5764
This API is implemented and ready to be integrated in UI (except the anonymous part)

The implementation problem is with personal link of a particular user showing which session a particular user has favourited. I did not see any issue for a user-specific link, and thus the API was not planned

For example, https://github.com/fossasia/open-event-frontend/issues/5763 this issue mentions `eventyay.com/e/8fa7fd14/schedule/my` as the URL, this is not user specific as it does not contain any user ID and thus, a user cannot share it with any other user to show which session they have favourited, and thus I did not create any public API for this

The above issue only mentions a separate section for viewing the sessions I liked, for which implementing a filter is the easiest way. 

All the issues mentioned related to `My Schedule` feature including https://github.com/fossasia/open-event-frontend/issues/5764 are still implementable. 

Sharing my schedule with someone else is not implementable as it was not mentioned in any of the above issues, and thus relevant APIs were not created.

So, right now we have:

- [x] Ability to add/remove sessions as favourite
- [x] Ability to see which sessions I favourited
- [x] Ability to see which sessions I'm a speaker in
- [ ] Ability to see who favourited a particular session (to be implemented)

I think all of the above cover what's mentioned and discussed in the My Schedule feature except marking a profile anonymous.

**Comment by mariobehling** [comment]
> The implementation problem is with personal link of a particular user showing which session a particular user has favourited. I did not see any issue for a user-specific link, and thus the API was not planned

Ok, we should have talked about this more in detail. Sched provides an option where users can click on an image of a person and see their entire schedule if they gave the permission to do that. This is why I mentioned the permissions in https://github.com/fossasia/open-event-frontend/issues/5764.

The public "my schedule" page should work as a kind of hub to see 
* the activity of a person in an event
* the profile
* the speaker sessions
* a contact button to get in touch or chat directly with the user.

So, the private "my schedule" should be the first steps towards this goal. Not everything must be possible right now. We already know that direct chatting is not so easy to set up, but the architecture should be done in a way that we can move in this direction.

Regarding the URLs you are right and we would still need to define what would be a public URL.

**Comment by mariobehling** [comment]
As we go with the filter for the start, please move the filter to the top right and of the central area and show two buttons a) Public Schedule and b) My Schedule. Make the active schedule darker.

Please also rename the heading "Sessions" to "Schedule".

**Comment by mariobehling** [comment]
Screenshot from 2021-02-14 11-23-13

**Comment by maze-runnar** [comment]
Screenshot from 2021-02-14 09-00-02

**Comment by iamareebjamal** [comment]
Add more padding to `Public Schedule` and `My Schedule`

File: app/templates/public.hbs [hunk 1]
```diff
@@ -40,9 +40,30 @@
         {{/if}}
       </div>
       {{#unless this.device.isMobile}}
-      <div class="ten wide column">
-        {{outlet}}
-      </div>
+        <div class="ten wide column">     
+          {{#if (eq this.session.currentRouteName 'public.sessions')}}
+            <div class="d-flex" style="align-items: center;">
+              <h1 class="ui header" id="session-heading">{{t 'Schedule'}}</h1>
```
Comment (iamareebjamal) [inline app/templates/public.hbs:46]: Why adding this in public? Why not in sessions.hbs?

**Comment by iamareebjamal** [comment]
Change `mySchedule` to `my_schedule`

And add the variable in sessions controller so `Public Schedule` is highlighted by default.

**Comment by maze-runnar** [comment]
Screenshot from 2021-02-15 16-15-42
Screenshot from 2021-02-15 16-15-37

**Pull request merged**